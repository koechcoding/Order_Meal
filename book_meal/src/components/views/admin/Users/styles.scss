
.show > .btn-secondary.dropdown-toggle {
    background-color: #008080 !important;
}
.dropdown-menu {
    margin-right: 80px !important;
}
.dropdown-item:hover,
.dropdown-item:focus,
.dropdown-item:active {
    cursor: pointer;
    color: black !important;
    background-color: transparent !important;
}