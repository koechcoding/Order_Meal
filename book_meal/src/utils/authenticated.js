export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}