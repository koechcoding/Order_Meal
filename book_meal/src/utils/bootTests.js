import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
