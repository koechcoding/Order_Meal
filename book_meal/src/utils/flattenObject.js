export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}