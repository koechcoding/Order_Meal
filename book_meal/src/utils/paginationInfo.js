export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}