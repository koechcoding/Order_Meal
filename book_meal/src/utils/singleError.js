const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;