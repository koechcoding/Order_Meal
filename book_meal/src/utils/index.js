import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}