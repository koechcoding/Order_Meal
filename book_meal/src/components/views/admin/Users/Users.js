import React from 'react';
import { Alert } from 'reactstrap';
import Filter from 'src/components/common/Filter';
import Content from 'src/components/common/Content';
import UsersTable from './components/UsersTable';
import ManageModal from './components/Manage';
import axios from 'src/axios';

import { singleError, paginationInfo } from 'src/utils';

class Users extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            page: 1,
            data: {},
            search: '',
            perPage: 5,
        }
    }

    componentWillMount() {
        this.fetchUsers()
    }

    fetchUsers = (config = {}) => {
        let { 
            page = this.state.page, 
            perPage = this.state.perPage,
            search = this.state.search, 
        } = config;
        this.props.setLoading(true);

        axios.auth();
        const link = `/users?page=${page}&search=${search}&per_page=${perPage}`;
        axios.get(link).then(({ data }) => {
            const pageInfo = paginationInfo(data);
            this.setState({
                ...this.state,
                page: pageInfo.currentPage,
                data,
            });
            this.props.setLoading(false);
            if (pageInfo.currentCount === 0 && pageInfo.currentPage !== 1) {
                this.fetchUsers({
                    page: pageInfo.currentPage - 1
                });
            }
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response,
            })
            this.props.setLoading(false);
        })
    }

    toggleManage = (e) => {
        this.setState({
            ...this.state,
            manageIsOpen: !this.state.manageIsOpen
        });
    }

    onFilter = (text) => {
        this.setState({
            ...this.state,
            search: text,
        });
        this.fetchUsers({search: text});
    }

    onToggle = (user) => {
        this.setState({
            ...this.state,
            manageIsOpen: true,
            toManage: user,
        });
    }

    onPageChange = (page) => {
        this.setState({
            ...this.state,
            page,
        });

        this.fetchUsers({
            page: page,
            search: this.state.search
        });
    }

    render() {
        const { 
            data,
            error,
            pageInfo,
            toManage,
            manageIsOpen
        } = this.state;

        const contentTop = (
            <div className="col-12 mb-2 pr-0 pr-sm-2">
                <h5 className="d-inline-block">Manage Users</h5>
            </div>
        );

        const contentFilter = (
            <Filter onFilter={this.onFilter} />
        );

        return (
            <Content 
                {...this.props}
                contentTop={contentTop} 
                contentFilter={contentFilter}>
            
                { error && <Alert color="danger"> { singleError(error) }</Alert> }
                     <UsersTable 
                         data={data}
                         pageInfo={pageInfo}
                         onPrev={this.onPageChange}
                         onToggle={this.onToggle}
                         onNext={this.onPageChange} />

                    <ManageModal 
                        {...this.props}
                        user={toManage} 
                        onChange={this.fetchUsers}
                        isOpen={manageIsOpen} 
                        toggle={this.toggleManage}/>
            </Content>
        );
    }
}

export default Users;