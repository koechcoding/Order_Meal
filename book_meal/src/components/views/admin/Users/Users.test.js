import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Users from './Users';
import Adapter from 'enzyme-adapter-react-16';

const props = {
    toggle: () => {},
    setLoading: () => {},
    toggleMenu: () => {},
    isOpen: false,
};
describe('<Users />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Users {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('sets current page', () => {
        const wrapper = shallow(<Users {...props}/>);
        wrapper.instance().onPageChange(2);
        expect(wrapper.state('page')).toBe(2);
    });
    it('toggles manage modal', () => {
        const wrapper = shallow(<Users {...props}/>);
        const prevState = wrapper.state('manageIsOpen');
        wrapper.instance().toggleManage();
        expect(wrapper.state('manageIsOpen')).toBe(!prevState);
    });
    it('shows manage modal', () => {
        const wrapper = shallow(<Users {...props}/>);
        wrapper.instance().onToggle({name: 'name'});
        expect(wrapper.state('toManage')).toEqual({name: 'name'});
    });
});