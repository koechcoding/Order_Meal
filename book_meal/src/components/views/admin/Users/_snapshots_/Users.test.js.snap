// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Users
    isOpen={false}
    setLoading={[Function]}
    toggle={[Function]}
    toggleMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        <UsersTable
          data={Object {}}
          onNext={[Function]}
          onPrev={[Function]}
          onToggle={[Function]}
          pageInfo={undefined}
/>,
        <ManageModal
          isOpen={undefined}
          onChange={[Function]}
          setLoading={[Function]}
          toggle={[Function]}
          toggleMenu={[Function]}
          user={undefined}
/>,
      ],
      "contentFilter": <Filter
        onFilter={[Function]}
/>,
      "contentTop": <div
        className="col-12 mb-2 pr-0 pr-sm-2"
>
        <h5
                className="d-inline-block"
        >
                Manage Users
        </h5>
</div>,
      "isOpen": false,
      "setLoading": [Function],
      "toggle": [Function],
      "toggleMenu": [Function],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {},
          "onNext": [Function],
          "onPrev": [Function],
          "onToggle": [Function],
          "pageInfo": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": undefined,
          "onChange": [Function],
          "setLoading": [Function],
          "toggle": [Function],
          "toggleMenu": [Function],
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          <UsersTable
            data={Object {}}
            onNext={[Function]}
            onPrev={[Function]}
            onToggle={[Function]}
            pageInfo={undefined}
/>,
          <ManageModal
            isOpen={undefined}
            onChange={[Function]}
            setLoading={[Function]}
            toggle={[Function]}
            toggleMenu={[Function]}
            user={undefined}
/>,
        ],
        "contentFilter": <Filter
          onFilter={[Function]}
/>,
        "contentTop": <div
          className="col-12 mb-2 pr-0 pr-sm-2"
>
          <h5
                    className="d-inline-block"
          >
                    Manage Users
          </h5>
</div>,
        "isOpen": false,
        "setLoading": [Function],
        "toggle": [Function],
        "toggleMenu": [Function],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {},
            "onNext": [Function],
            "onPrev": [Function],
            "onToggle": [Function],
            "pageInfo": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": undefined,
            "onChange": [Function],
            "setLoading": [Function],
            "toggle": [Function],
            "toggleMenu": [Function],
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;